src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -286,18 +287,20 @@ void OTARequestor::Reset()
     StoreCurrentUpdateInfo();
 }
 
-EmberAfStatus OTARequestor::HandleAnnounceOTAProvider(app::CommandHandler * commandObj,
-                                                      const app::ConcreteCommandPath & commandPath,
-                                                      const AnnounceOtaProvider::DecodableType & commandData)
+void OTARequestor::HandleAnnounceOTAProvider(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+                                             const AnnounceOtaProvider::DecodableType & commandData)
 {
+    VerifyOrReturn(commandObj != nullptr, ChipLogError(SoftwareUpdate, "Invalid commandObj, cannot handle AnnounceOTAProvider"));
+
     auto & announcementReason = commandData.announcementReason;
 
     ChipLogProgress(SoftwareUpdate, "OTA Requestor received AnnounceOTAProvider");
 
-    if (commandObj == nullptr || commandObj->GetExchangeContext() == nullptr)
+    if (commandObj->GetExchangeContext() == nullptr)
```

Comment:
This block really doesn't make sense.  If it has no exchange context, what is the point of calling AddStatus on it?  It can't send the status....

Reply:
So... do nothing then? I can just make this return.

Reply:
That seems fine, yes.  But also, this _really_ shouldn't be getting called with a CommandHandler that has no exchange context....

Reply:
I kind of thought about removing this check altogether since it also isn't specifically being used in this function...

Reply:
Just going to remove this check